src/Analyzers/CSharp/Analyzers/SimplifyLinqExpressions/CSharpSimplifyLinqExpressionsDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,158 @@
+            {
+                return;
+            }
+
+            AddIfNotNull(whereMethods, enumMethods);
+
+            // add all valid linq calls
+            foreach (var id in s_validLinqCalls)
+            {
+                enumMethods = enumNamedType?.GetMembers(id).OfType<IMethodSymbol>();
+                AddIfNotNull(linqMethods, enumMethods);
+            }
+
+            if (whereMethods.Count > 0 && linqMethods.Count > 0)
+            {
+                context.RegisterOperationAction(
+                    context => AnalyzeAction(
+                        context, whereMethods.ToImmutable(), linqMethods.ToImmutable()), OperationKind.Invocation);
+            }
+
+            return;
+
+            static void AddIfNotNull(ImmutableHashSet<IMethodSymbol>.Builder set, IEnumerable<IMethodSymbol>? others)
+            {
+                if (others != null)
+                {
+                    set.UnionWith(others);
+                }
+            }
+        }
+
+        public void AnalyzeAction(OperationAnalysisContext context, ImmutableHashSet<IMethodSymbol> whereMethods, ImmutableHashSet<IMethodSymbol> linqMethods)
+        {
+            var invocationOperation = (IInvocationOperation)context.Operation;
+            var semanticModel = context.Operation.SemanticModel;
+
+            var firstChildArgument = invocationOperation.Children.FirstOrDefault(c => c is IArgumentOperation);
+
+            if (firstChildArgument is null)
+            {
+                return;
+            }
+
+            var whereClause = firstChildArgument.Children.FirstOrDefault(c => c is IInvocationOperation);
+
+            if (whereClause is null)
+            {
+                return;
+            }
+
+            var isLambda = whereClause.Children.Any(c => c.Syntax is ArgumentSyntax argument && argument.Expression is SimpleLambdaExpressionSyntax);
+
+            if (!isLambda)
+            {
+                return;
+            }
+
+            if (whereClause is IInvocationOperation method && !whereMethods.Contains(method.TargetMethod.OriginalDefinition))
+            {
+                return;
+            }
+
+            // check that the Where clause is followed by a call to a valid method i.e. one of First, FirstOrDefault, Single, SingleOrDefault, etc..
+            // and that it is also not user defined
+            var invocationDefinition = invocationOperation.TargetMethod.OriginalDefinition;
+
+            if (!linqMethods.Contains(invocationDefinition))
+            {
+                return;
+            }
+
+            // Check that the Where clause is followed by a call with no predicate
+            var arguments = invocationOperation.TargetMethod.Parameters;
+            if (arguments.IsEmpty)
+            {
+                return;
+            }
+
+            var operation = context.Operation;
+            var node = operation.Syntax;
+
+            // check that the nodes expression is a MemberAccessExpressionSyntax
```

Comment:
this isn't a helpful comment.  it just describes teh check that is happening.  It would be better to give an actual example of what we're looking for.  i.e.